### PR TITLE
task autoupdater, disabled in conf

### LINKED ISSUE #26: Task for auto updater
Create a new tasks that:

1. Periodically checks if a new version of srvcheck is available
2. If true, the pip install the new version and send a notification
3. Restart the service

This task should be disabled by default; it could be an useful tool during development.

### PR DESCRIPTION
**Pull request by AlessandroPaparella**
Closes #26

### CODE REVIEW

File: srvcheck/tasks/taskautoupdater.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+class TaskAutoUpdater(Task):
+    def __init__(self, conf, notification, system, chain):
+        super().__init__('TaskAutoUpdater', conf, notification, system, chain, minutes(60), hours(2))
+    
+    def run(self):
+        nTag = requests.get('https://api.github.com/repos/openbitlab/srvcheck/git/refs/tags').json()[-1]['ref'].split('/')[-1]
+        if self.conf["version"]["tagVersion"] != nTag:
+            self.notify('Installing new monitor version: %s' % (nTag))
+            Bash('pip install --upgrade git+https://github.com/openbitlab/srvcheck')
```
Comment (dakk) [inline srvcheck/tasks/taskautoupdater.py:15]: Doing this we install the latest commit: we need instead to specifiy the tag to install using @tagname suffix (ie: 'git+https://github.com/openbitlab/srvcheck@v0.1')

**Review by dakk — CHANGES_REQUESTED** [review]

File: srvcheck/tasks/taskautoupdater.py [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
+from ....srvcheck import srvcheck
 
 class TaskAutoUpdater(Task):
     def __init__(self, conf, notification, system, chain):
         super().__init__('TaskAutoUpdater', conf, notification, system, chain, minutes(60), hours(2))
     
     def run(self):
-        nTag = requests.get('https://api.github.com/repos/openbitlab/srvcheck/git/refs/tags').json()[-1]['ref'].split('/')[-1]
-        if self.conf["version"]["tagVersion"] != nTag:
+        nTag = requests.get('https://api.github.com/repos/openbitlab/srvcheck/git/refs/tags').json()[-1]['ref'].split('/')[-1].split('v')[1]
```
Comment (dakk) [inline srvcheck/tasks/taskautoupdater.py:13]: ```[1:]```

**Review by franz-ops — COMMENTED** [review]
"import subprocess" in taskautoupdater.py can be deleted

**Review by dakk — CHANGES_REQUESTED** [review]

File: srvcheck/tasks/taskautoupdater.py (hunk 2 above)
Comment (dakk) [reply srvcheck/tasks/taskautoupdater.py:13]: se riesci, ristabilisci i test che falliscono

Comment (dakk) [reply srvcheck/tasks/taskautoupdater.py:13]: also fix failing tests

**Comment by AlessandroPaparella** [comment]
> "import subprocess" in taskautoupdater.py can be deleted

done

File: srvcheck/tasks/taskautoupdater.py (hunk 2 above)
Comment (AlessandroPaparella) [reply srvcheck/tasks/taskautoupdater.py:11]: > also fix failing tests

solved

**Pull request merged**